server/datastore/mysql/hosts.go

```diff
@@ -848,6 +848,14 @@ func (ds *Datastore) ListHosts(ctx context.Context, filter fleet.TeamFilter, opt
 
 	sql += hostMDMSelect
 
+	if opt.ListOptions.OrderKey == "last_restarted" {
+		sql += /*`,
+			-FLOOR(UNIX_TIMESTAMP() * 1000 - (((UNIX_TIMESTAMP() - UNIX_TIMESTAMP(detail_updated_at)) * 1000) + (uptime / 1000000))) AS last_restarted
+				`*/`,
+		DATE_SUB(detail_updated_at, INTERVAL uptime/1000 MICROSECOND) as last_restarted
```

Comment:
We need to account for the case where `detail_updated_at` is `NULL`.

Reply:
@lucasmrod seems like having `detail_updated_at` as NULL returns a 500 error even without filtering on `last_restarted`. I fixed that by adding an additional `COALESCE` to line 806. Is it OK to fix that technically separate issue here? I added it because even with the `COALESCE` on line 853, the endpoint still breaks if `detail_updated_at` is NULL